### ISSUE TITLE
Perform a11y audits of ListBox-related Components

### ISSUE THREAD

**Issue by joshblack** [IBM/carbon-components-react#468]
With more of the ListBox-related components being merged in, we should take the time to do proper a11y audits on these components. While downshift provides a good number of the a11y requirements we need to hit, I'm sure that I forgot some while coding, or am simply unaware of the best practices around a11y for some of the controls we've created.

This issue is to conduct the audit of the components, and then create an issue for the implementation for anything that needs updating.
